### PR TITLE
Feature/update fundation

### PR DESCRIPTION
**Pull request by john-ko**

### CODE REVIEW

File: middleware/controllers.js [hunk 1]
```diff
@@ -26,83 +103,15 @@ module.exports = async function(app, fundation) {
...
   app.set('case sensitive routing', true);
 
-  // Example: app.r.get('/test', function (req, res, next) {
-  // app.r = express.Router({
-  //   strict: true,
-  //   caseSensitive: true
-  // });
-
-  // app.use(app.r);
+  app.renderVue = renderVue
```
Comment (john-ko) [inline middleware/controllers.js:125]: maybe use `app.locals.renderVue` instead?

File: middleware/controllers.js [hunk 2]
```diff
@@ -18,6 +20,88 @@ var express = require('express')
...
+
+        // return generic error
+        return res.status(500).end('Internal Error 500')
+      }
+
+      const m = context.meta.inject()
+      let vueMeta = m.meta.text() + m.title.text() + m.link.text() + m.style.text() + m.script.text() + m.noscript.text()
+      let prependHeadTag = ''
+      if (app.prependToHeadTag && app.prependToHeadTag.length) {
+        app.prependToHeadTag.map(function prependToHeadTag (head) {
```
Comment (john-ko) [inline middleware/controllers.js]: better name?

File: middleware/controllers.js [hunk 3]
```diff
@@ -18,6 +20,90 @@ var express = require('express')
...
+      }
+
+      const m = context.meta.inject()
+      let vueMeta = m.meta.text() + m.title.text() + m.link.text() + m.style.text() + m.script.text() + m.noscript.text()
+      let prependHeadTag = ''
+      if (app.prependToHeadTag && app.prependToHeadTag.length) {
+        app.prependToHeadTag.map(function (head) {
+          prependHeadTag += head()
+        })
+      }
```
Comment (john-ko) [inline middleware/controllers.js]: thinking of using something like this
```
const prependHeadTag = _.get(app, 'prependToHeadTag', []).reduce((accumilator, callback) => {
  return accumilator + callback()
}, '')
```
or remove all together?

**Comment by seemo66** [comment]
Code looks good.

**Pull request closed without merging**